SYCL/ESIMD/lsc/lsc_usm_store_u32.cpp

```diff
@@ -31,6 +31,11 @@ template <int TestCastNum, typename T> bool tests() {
   passed &= test<TestCastNum + 9, T, 2, 2, 1, 16, true>();
   passed &= test<TestCastNum + 10, T, 4, 4, 1, 4, true>();
 
+  // large number of elements
+  passed &= test<TestCastNum + 11, T, 4, 4, 1, 128, true,
```

Comment:
128 is correct only for T=int32_t, and is too much for T=int64_t.
Please use some checks, i.e. 'if constexpr (sizeof(T) <= 4)'

Can you please verify 256*int16_t and 512*int8_t ?

Finally, please ensure that data is allocated with proper alignment.